source/common/http/header_map_impl.cc

```diff
@@ -564,6 +566,12 @@ HeaderMap::Lookup HeaderMapImpl::lookup(const LowerCaseString& key,
   }
 }
 
+void HeaderMapImpl::clear() {
+  memset(&inline_headers_, 0, sizeof(inline_headers_));
```

Comment:
nit: this looks scary on its own. Can you instead write:
```
inline_headers_.clear();
```
and put
```
void clear() { memset(this, 0, sizeof(*this)); }
```
into the struct for AllInlineHeaders in header_map_impl.h? There you can see that these are all pointers and safe to clear.

Reply:
+1

Reply:
Done.